build/bin/write-dockerfile.sh

```diff
@@ -15,17 +15,14 @@ SAGE_ROOT=.
 export PATH="$SAGE_ROOT"/build/bin:$PATH
 SYSTEM_PACKAGES=$EXTRA_SYSTEM_PACKAGES
 CONFIGURE_ARGS="--enable-option-checking "
-for PKG_BASE in $(sage-package list --has-file=distros/$SYSTEM.txt $SAGE_PACKAGE_LIST_ARGS) $EXTRA_SAGE_PACKAGES; do
-    PKG_SCRIPTS="$SAGE_ROOT"/build/pkgs/$PKG_BASE
-    if [ -d $PKG_SCRIPTS ]; then
-       SYSTEM_PACKAGES_FILE=$PKG_SCRIPTS/distros/$SYSTEM.txt
-       PKG_SYSTEM_PACKAGES=$(echo $(${STRIP_COMMENTS} $SYSTEM_PACKAGES_FILE))
-       if [ -n "PKG_SYSTEM_PACKAGES" ]; then
-           SYSTEM_PACKAGES+=" $PKG_SYSTEM_PACKAGES"
-           if [ -f $PKG_SCRIPTS/spkg-configure.m4 ]; then
-               CONFIGURE_ARGS+="--with-system-$PKG_BASE=${WITH_SYSTEM_SPKG} "
-           fi
-       fi
+for SPKG in $(sage-package list $SAGE_PACKAGE_LIST_ARGS) $EXTRA_SAGE_PACKAGES; do
+    SYSTEM_PACKAGE=$(sage-get-system-packages $SYSTEM $SPKG)
+    if [ -n "${SYSTEM_PACKAGE}" ]; then
+	# SYSTEM_PACKAGE can be empty if, for example, it corresponds
+	# to a python package and --enable-system-site-packages was
+	# not passed to ./configure.
```

Comment:
This script is not invoked within a Sage environment -- it would be better to refer to the environment variable that controls it.

Reply:
fixed